test/testdata/rbi/with_without__3.rb

```diff
@@ -0,0 +1,3 @@
+# typed: true
+
+T.reveal_type(MyGem.method_thats_part_of_my_public_api) # error: Revealed type: `NilClass`
```

Comment:
For this file and the two rbis, is it expected that they're all present for one tests, or are they each different tests?

Reply:
the `foo__N.rb` syntax is a thing the test corpus runner already was aware of prior to this change. it's a way to make a single bazel test target that contains multiple ruby files.

in this change, I added the ability to intermix `foo__N.rbi` files with the `.rb` files.

Reply:
e.g., this existing test:

https://github.com/sorbet/sorbet/blob/d925413dff03c1b58d390b8d73f53fe13da8fdc8/test/testdata/infer/aliases__1.rb
https://github.com/sorbet/sorbet/blob/d925413dff03c1b58d390b8d73f53fe13da8fdc8/test/testdata/infer/aliases__2.rb

is one bazel target for a test that contains code in two files.

Reply:
Ah, nice!